src/main/java/data/Vendor.java

```diff
@@ -132,4 +140,27 @@ public String buildJsonConfig() {
     config += "]}";
     return config;
   }
+
+  /**
+   * Updates the Account specified in the HttpServletRequest.
+   * @param request a request containing form data to edit an existing account
+   */
+  public void editVendorAccount(HttpServletRequest request) {
+    String accountId = request.getParameter(FormIdNames.ACCOUNT_ID);
+    Account account = getAccountById(accountId);
+
+    // Checked previously to see if the account exists, but this confirms
+    // that assumption.
+    assert account != null;
```

Comment:
Will this throw an error if account is null? If so, you might want to be able to handle that somewhere.

Reply:
I decided to remove it for now. The error it will throw when this happens is sort of designed to crash the program, but there's a check elsewhere to make sure this method isn't called on an empty account